user_service/README.md

```diff
@@ -5,58 +5,119 @@ Welcome to the User Service API documentation. This API provides endpoints for m
 ## Table of Contents
 
 1. [Endpoints](#endpoints)
-4. [Running the Backend Service](#running-the-backend-service)
-5. [Environment Variables and .env File](#environment-variables-and-env-file)
+2. [User Authentication and Authorization](#user-authentication-and-authorization)
+3. [Running the Backend Service](#running-the-backend-service)
+4. [Environment Variables and .env File](#environment-variables-and-env-file)
+
+
+## User Authentication and Authorization
+To access certain endpoints, users need to be authenticated and authorized. The User Service uses JWT (JSON Web Tokens) for user authentication. To authenticate, include a valid JWT token in the request headers.
+
+Header: Authorization: Bearer <access_token>
+Replace <access_token> with a valid access token obtained by logging in using the /api/login endpoint. Access tokens contain information about the user's name and roles.
 
 ## Endpoints
 
 ### Note that the user service runs on port 8000.
 
-### Create User
+### Register User
 
-- **URL**: `/api/users`
+- **URL**: `/api/register/`
 - **Method**: `POST`
 - **Description**: Create a new user.
 - **Request Body**:
   ```json
   {
-    "name": "john_doe",
+    "username": "john_doe",
+    "password": "pwd",
+    (Optional)"displayed_name": "name"
   }
 - **Response**:
   - 201 Created on success with user details
+  - 400 Bad Request on failure
+
+### Login User
+- **URL**: `/api/login/`
+- **Method**: `POST`
+- **Description**: Login as a user.
+- **Request Body**:
+  ```json
+  {
+    "username": "john_doe",
+    "password": "pwd"
+  }
+- **Response**:
+  - 200 on success with access_token and refresh_token attached
+  - 401 on failure
+
+
+### Obtain new access token with refresh token
+- **URL**: `/api/token/refresh/`
+- **Method**: `POST`
+- **Description**: Obtain new access token.
+- **Request Body**:
+  ```json
+  {
+    "refresh": "213245tygdfs...token...."
+  }
+- **Response**:
+  - new access token on success
+
+
+### Verify the validity of a token
+- **URL**: `/api/token/verify/`
+- **Method**: `POST`
+- **Description**: Verify a token.
+- **Request Body**:
+  ```json
+  {
+    "token": "213245tygdfs...token...."
+  }
+- **Response**:
+  - 200 on valid token
```

Comment:
does this return a response for invalid tokens?

Reply:
yeah it will indicate if it is an invalid token, see the example below:

<img width="751" alt="image" src="https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g34/assets/89072240/f82ad7b3-3131-4f99-8698-f69e1ea3927d">


i will update the readme later

Reply:
Also the token will have the following information:
<img width="862" alt="image" src="https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g34/assets/89072240/0c6ea349-4d94-4467-97b7-492c48324b65">

so that the question service can decrypt it and obtain the user_role information